### DISCUSSION TITLE
How to overload +compile to execute task before it?

### DISCUSSION THREAD

**Discussion by SuperFola** [sbt/sbt#7518] [Q&A]
Hi,

I've successfully overloaded compile and package to execute a custom task before (download a swagger, to be used by open api spec generator by eikek), but I can't seem to find a way to overload +compile or +package, since they seem to have been both implemented in the command parser, instead of as plain sbt task keys.

```scala
object CustomPlugin extend AutoPlugin {
    // ...
    override def projectSettings =
    Seq(
      copyResourcesFromJarsSetting,
      compile := Def.sequential(copyResourcesFromJars, Compile / compile).value,
      `package` := Def.sequential(copyResourcesFromJars, Compile / `package`).value,
      // ...
    )
}
```

**Comment by eed3si9n** [comment]
I think `Def.sequential(...)` is the wrong way of generating a resource. Plz see https://www.scala-sbt.org/1.x/docs/Howto-Generating-Files.html#Generate+resources.

**Reply by SuperFola** [reply]
Thanks for the pointer

I've been trying to apply this, however it seems we are stuck with having resources generated after sources using generators (while I'm looking for a resource to be created first, so that I can generate sources from it later (managed by another plugin, sbt-openapi-schema)).

Having read the doc you posted, it seems that what I want to achieve isn't possible?
